### DISCUSSION TITLE
Is there a way to access apps using local IP?

### DISCUSSION THREAD

**Discussion by dhanadhan** [caprover/caprover#1626] [Q&A]
I successfully deployed Caprover locally and configured local dnsmasq on Ubuntu. I am using Cloudflare tunnel to route local traffic on a domain. I can successfully access caprover admin dashboard through cloudflare since I provided them "LOCALIP:3000". 
I deployed an app, and it is successfully running locally at "app.domain.local". However, when I provide "app.domain.local" to Cloudflare, I can't access it. Is there a way to access the app using the local IP? Something like "app.192.168.1.2"? Is there anything I can do to get this working through Cloudflare tunnel? I don't want to add the domain directly to Caprover since I have to expose my public IP which is not static. 

Thanks

**Comment by githubsaturn** [comment]
> Is there a way to access the app using the local IP? Something like "app.192.168.1.2"? 

No

> Is there anything I can do to get this working through Cloudflare tunnel? 

Not sure how cloudflare works. It's really; a question for them. CapRover simply sets up your nginx and after that there is nothing CapRover specific about your set up. My guess is that your cloudflare tunnel doesn't resolve the DNS. It's something that you have to follow up with them to see what's causing this.

**Reply by dhanadhan** [reply]
I will try to follow with them. Thank you.

**Comment by dhanadhan** [comment]
If anyone else is wondering about an alternative way to externally access local apps running on Caprover.

I can't believe I figured it out. I installed UptimeKuma app just to check it out, and I saw it allows you to run Reverse Proxy through Cloudflare tunnel. I created a new Tunnel on Cloudflare and added the token to UptimeKuma. Then added local app url (srv-captain--app-name) generated by Caprover to Cloudflare. I can now successfully access an app on my domain (https) through Cloudflare tunnel :)

**Reply by rafiqlightwala** [reply]
Hi @dhanadhan can you please lay down in simple steps how to shift a current app with a domain like app1.apps.domain.com to cloudflare tunnel without having any downtime?
